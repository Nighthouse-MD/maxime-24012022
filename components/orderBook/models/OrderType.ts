enum OrderType {
    ask = "sell",
    bid = "buy"
}

export default OrderType;