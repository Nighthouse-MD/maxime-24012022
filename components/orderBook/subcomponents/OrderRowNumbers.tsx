import { memo } from 'react';
import { Col } from 'react-bootstrap';
import styles from '../OrderBook.module.scss'
import OrderType from '../models/OrderType';

const OrderRowNumbers = ({ type, price, quantity, cumulativeVolume }: OrderRowNumbersProps) => {
    return <>
        <Col xs={{ order: 1 }} md={{ order: type === OrderType.bid ? 3 : 1 }} className={`${styles.price} ${type === OrderType.ask ? styles.red : styles.green}`}>
            <span className="align-middle">{price}</span>
        </Col>
        <Col xs={{ order: 2 }} md={{ order: 2 }} className={styles.quantity}>
            <span className="align-middle">{quantity}</span>
        </Col>
        <Col xs={{ order: 3 }} md={{ order: type === OrderType.bid ? 1 : 3 }} className={styles.quantity}>
            <span className="align-middle">{cumulativeVolume}</span>
        </Col>
    </>
};

interface OrderRowNumbersProps {
    type: OrderType;
    price: string;
    quantity: string;
    cumulativeVolume: string;
}

export default memo(OrderRowNumbers);