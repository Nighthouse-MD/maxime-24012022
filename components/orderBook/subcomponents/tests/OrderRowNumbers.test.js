import OrderRowNumbers from '../OrderRowNumbers';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import OrderType from '../../models/OrderType';

describe('OrderRowNumbers', () => {
    it('shows the correct order for OrderType.ask', () => {
        render(<OrderRowNumbers type={OrderType.ask} price={123} quantity={234} cumulativeVolume={567} />);

        const priceElement = screen.queryByText("123");
        expect(priceElement).toBeInTheDocument();
        expect(priceElement.parentElement.className).toBe('price red order-md-1 order-1 col');

        const sizeElement = screen.queryByText("234");
        expect(sizeElement).toBeInTheDocument();
        expect(sizeElement.parentElement.className).toBe('quantity order-md-2 order-2 col');

        const totalElement = screen.queryByText("567");
        expect(totalElement).toBeInTheDocument();
        expect(totalElement.parentElement.className).toBe('quantity order-md-3 order-3 col');
    });

    it('shows the correct order for OrderType.bid', () => {
        render(<OrderRowNumbers type={OrderType.bid} price={123} quantity={234} cumulativeVolume={567} />);

        const priceElement = screen.queryByText("123");
        expect(priceElement).toBeInTheDocument();
        expect(priceElement.parentElement.className).toBe('price green order-md-3 order-1 col');

        const sizeElement = screen.queryByText("234");
        expect(sizeElement).toBeInTheDocument();
        expect(sizeElement.parentElement.className).toBe('quantity order-md-2 order-2 col');

        const totalElement = screen.queryByText("567");
        expect(totalElement).toBeInTheDocument();
        expect(totalElement.parentElement.className).toBe('quantity order-md-1 order-3 col');
    });

    it('renders correctly for OrderType.ask', () => {
        const tree = renderer
            .create(<OrderRowNumbers type={OrderType.ask} price={123} quantity={234} cumulativeVolume={567} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly for OrderType.bid', () => {
        const tree = renderer
            .create(<OrderRowNumbers type={OrderType.bid} price={123} quantity={234} cumulativeVolume={567} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});