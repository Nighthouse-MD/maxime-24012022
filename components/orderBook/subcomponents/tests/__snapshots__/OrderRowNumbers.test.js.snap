// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderRowNumbers renders correctly for OrderType.ask 1`] = `
Array [
  <div
    className="price red order-md-1 order-1 col"
  >
    <span
      className="align-middle"
    >
      123
    </span>
  </div>,
  <div
    className="quantity order-md-2 order-2 col"
  >
    <span
      className="align-middle"
    >
      234
    </span>
  </div>,
  <div
    className="quantity order-md-3 order-3 col"
  >
    <span
      className="align-middle"
    >
      567
    </span>
  </div>,
]
`;

exports[`OrderRowNumbers renders correctly for OrderType.bid 1`] = `
Array [
  <div
    className="price green order-md-3 order-1 col"
  >
    <span
      className="align-middle"
    >
      123
    </span>
  </div>,
  <div
    className="quantity order-md-2 order-2 col"
  >
    <span
      className="align-middle"
    >
      234
    </span>
  </div>,
  <div
    className="quantity order-md-1 order-3 col"
  >
    <span
      className="align-middle"
    >
      567
    </span>
  </div>,
]
`;
